pkg/suggestion/nasrl_service.py

```diff
@@ -1,22 +1,227 @@
+        organized_arc = [0 for _ in range(self.num_layers)]
+        record = 0
+        for l in range(self.num_layers):
+            organized_arc[l] = arc[record: record + l + 1]
+            record += l + 1
+
+        nn_config = dict()
+        nn_config['num_layers'] = self.num_layers
+        nn_config['input_size'] = self.input_size
+        nn_config['output_size'] = self.output_size
+        nn_config['embedding'] = dict()
+        for l in range(self.num_layers):
+            opt = organized_arc[l][0]
+            nn_config['embedding'][opt] = self.search_space[opt].get_dict()
+
+        organized_arc_json = json.dumps(organized_arc)
+        nn_config_json = json.dumps(nn_config)
+
+        organized_arc_str = str(organized_arc_json).replace('\"', '\'')
+        nn_config_str = str(nn_config_json).replace('\"', '\'')
+
+        print("\nNew Neural Network Architecture (internal representation):")
+        print(organized_arc_json)
+        print("\nCorresponding Seach Space Description:")
+        print(nn_config_str)
+        print()
+
         trials = []
+        trials.append(api_pb2.Trial(
+                study_id=request.study_id,
+                parameter_set=[
+                    api_pb2.Parameter(
+                        name="architecture",
+                        value=organized_arc_str,
+                        parameter_type= api_pb2.CATEGORICAL),
+                    api_pb2.Parameter(
+                        name="nn_config",
+                        value=nn_config_str,
+                        parameter_type= api_pb2.CATEGORICAL)
+                ], 
+            )
+        )
 
+        channel = grpc.beta.implementations.insecure_channel(self.manager_addr, self.manager_port)
+        with api_pb2.beta_create_Manager_stub(channel) as client:
+            for i, t in enumerate(trials):
+                ctrep = client.CreateTrial(api_pb2.CreateTrialRequest(trial=t), 10)
+                trials[i].trial_id = ctrep.trial_id
+                self.current_trial_id = ctrep.trial_id
+            print("Trial {} Created\n".format(ctrep.trial_id))
+            self.current_trial_id = ctrep.trial_id
+        
+        self.ctrl_step += 1
         return api_pb2.GetSuggestionsReply(trials=trials)
+
+    def GetEvaluationResult(self, studyID):
+        worker_hist = []
+        channel = grpc.beta.implementations.insecure_channel(self.manager_addr, self.manager_port)
+        with api_pb2.beta_create_Manager_stub(channel) as client:
+            gwfrep = client.GetWorkerFullInfo(api_pb2.GetWorkerFullInfoRequest(study_id=studyID, trial_id=self.current_trial_id, only_latest_log=True), 10)
+            worker_hist = gwfrep.worker_full_infos
+
+        for w in worker_hist:
+            if w.Worker.status == api_pb2.COMPLETED:
+                for ml in w.metrics_logs:
+                    print("Evaluation result of previous candidate:", ml.values[-1].value)
+
+        return float(ml.values[-1].value)
```

Comment:
I suggest keep line 182 same indent with line 179, had better handle "no worker status == api_pb2.COMPLETED" case (although we except at least one worker changing to completed trigger this method).